src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -1491,5 +1557,105 @@ protected void Log(string msg, WarningLevel severity)
+                    nodeModel.UpdateValue(new UpdateValueParams("IsVisible", nodeViewInfo.IsVisible.ToString()));
+                    nodeModel.UpdateValue(new UpdateValueParams("IsUpstreamVisible", nodeViewInfo.IsUpstreamVisible.ToString()));
+                }
+            }
+
+            foreach (ExtraNoteViewInfo noteViewInfo in workspaceViewInfo.Notes)
+            {
+                var guidValue = IdToGuidConverter(noteViewInfo.Id);
+
+                // TODO, QNTM-1099: Figure out if ZIndex needs to be set here as well
+                var noteModel = new NoteModel(noteViewInfo.X, noteViewInfo.Y, noteViewInfo.Text, guidValue);
+                this.AddNote(noteModel);
+            }
+
+            foreach (ExtraAnnotationViewInfo annotationViewInfo in workspaceViewInfo.Annotations)
+            {
+                // TODO, QNTM-1099: Determine where to set the ID for annotations
+                // Guid guidValue = IdToGuidConverter(annotationInfo.Id);
+
+                // Create a collection of nodes in the given annotation
+                var nodes = new List<NodeModel>();
+                foreach (string nodeId in annotationViewInfo.Nodes)
+                {
+                  var guidValue = IdToGuidConverter(nodeId);
+                  if (guidValue == null)
+                    continue;
+
+                  // NOTE: Some nodes may be annotations and not be found here
+                  var nodeModel = Nodes.FirstOrDefault(node => node.GUID == guidValue);
+                  if (nodeModel == null)
+                    continue;
+
+                  nodes.Add(nodeModel);
+                }
+
+                // Create a collection of notes in the given annotation
+                var notes = new List<NoteModel>();
+                foreach (string noteId in annotationViewInfo.Nodes)
+                {
+                  var guidValue = IdToGuidConverter(noteId);
+                  if (guidValue == null)
+                    continue;
+
+                  // NOTE: Some nodes may not be annotations and not be found here
+                  var noteModel = Notes.FirstOrDefault(note => note.GUID == guidValue);
+                  if (noteModel == null)
+                    continue;
+
+                  notes.Add(noteModel);
+                }
+
+                var annotationModel = new AnnotationModel(nodes, notes);
+                annotationModel.AnnotationText = annotationViewInfo.Title;
+                annotationModel.FontSize = annotationViewInfo.FontSize;
+                annotationModel.Background = annotationViewInfo.Background;
+                this.AddNewAnnotation(annotationModel);
+            }
+
+            // TODO, QNTM-1099: These items are not in the extra view info
+            // Name = info.Name;
+            // Description = info.Description;
+            // FileName = info.FileName;
+        }
+
+        private Guid IdToGuidConverter(string id)
```

Comment:
It is possible to use the utility class. However, this code was copied (and altered) from the IdToGuidConverter class in SerializationConverters.cs file that you referenced above. It was copied because there is a string input here instead of a JSON reader object. 

I did not use the utility function because the referenced code logs to the console when the input is not a GUID, and the utility method does not do any logging. 

I can leave it as it is and have logging, or switch to the utility function and have no logging. It should be possible to use the utility function here and figure out how to do the logging if needed, but it seemed to me like kind of a twisty way to go to reuse just a few lines of code.